pkg/controller/cluster/pools.go

```diff
@@ -102,7 +102,11 @@ func poolSSMatchesSpec(tenant *miniov2.Tenant, pool *miniov2.Pool, ss *appsv1.St
 	if !reflect.DeepEqual(expectedMetadata.Labels, ss.ObjectMeta.Labels) {
 		poolMatchesSS = false
 	}
-	if !reflect.DeepEqual(expectedMetadata.Annotations, ss.ObjectMeta.Annotations) {
+	expectedAnnotations := expectedMetadata.Annotations
+	currentAnnotations := ss.ObjectMeta.Annotations
+	delete(expectedAnnotations, "kubectl.kubernetes.io/last-applied-configuration")
+	delete(currentAnnotations, "kubectl.kubernetes.io/last-applied-configuration")
+	if !reflect.DeepEqual(expectedAnnotations, currentAnnotations) {
```

Comment:
is this one that is causing restarts? @Alevsk

Reply:
```
const k8sAnnotationLastApplied = "kubectl.kubernetes.io/last-applied-configuration"
```

Reply:
Correct @harshavardhana , this is also preventing the rollout restart of the MinIO statefulsets to be completed, I'm not editing the original annotations since this one is controlled by `kube-controller`, I'm just skipping it from the comparison

Reply:
Correct let's keep the string as a constant.

Reply:
@harshavardhana done